lib/compile/csdl2openapi.js

```diff
@@ -106,7 +106,60 @@ module.exports.csdl2openapi = function (
     if (externalDocs && Object.keys(externalDocs).length > 0) {
         openapi.externalDocs = externalDocs;
     }
+    const extensions = getExtensions(csdl, 'root');
+    if (extensions && Object.keys(extensions).length > 0) {
+        Object.assign(openapi, extensions);
+    }
+
+    // function to read @OpenAPI.Extensions and get them in the generated openAPI document
+    function getExtensions(csdl, level) {
+        let extensionObj = {};
+        let containerSchema = {};
+        if (level ==='root'){
+        const namespace = csdl.$EntityContainer ? nameParts(csdl.$EntityContainer).qualifier : null;
+        containerSchema = csdl.$EntityContainer ? csdl[namespace] : {};
+        }
+        else if(level === 'schema' || level === 'operation'){
+            containerSchema = csdl;
+        }
+
+        for (const [key, value] of Object.entries(containerSchema)) {
+            if (key.startsWith('@OpenAPI.Extensions')) {
+                const annotationProperties = key.split('@OpenAPI.Extensions.')[1];
+                const keys = annotationProperties.split('.');
```

Comment:
Processing for fields in `Defined Specification Extensions` like `x-sap-stateinfo` needs to be implemented.